context/src/main/java/io/grpc/Context.java

```diff
@@ -895,8 +920,16 @@ public String toString() {
     public abstract void detach(Context toDetach, Context toRestore);
 
     /**
-     * Implements {@link io.grpc.Context#current}.  Returns the context of the current scope.
+     * Implements {@link io.grpc.Context#current}.
+     *
+     * <p>Caution: {@link Context} interprets a return value of {@code null} to mean the same
```

Comment:
maybe make a test that via reflection changes Storage to something that returns null? then test that ROOT is substituted